bee-network/src/service/service.rs

```diff
@@ -0,0 +1,486 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    command::{Command, CommandReceiver, CommandSender},
+    event::{Event, EventSender, InternalEvent, InternalEventReceiver, InternalEventSender},
+};
+
+// TODO: introduce `service::error` module.
+use crate::{
+    alias,
+    init::global::reconnect_interval_secs,
+    peer::{error::Error as PeerError, list::PeerListWrapper as PeerList},
+    types::{PeerInfo, PeerRelation},
+};
+
+use bee_runtime::shutdown_stream::ShutdownStream;
+
+use futures::{channel::oneshot, StreamExt};
+use libp2p::{identity, Multiaddr, PeerId};
+use log::*;
+use rand::Rng;
+use tokio::time::{self, Duration, Instant};
+use tokio_stream::wrappers::{IntervalStream, UnboundedReceiverStream};
+
+pub struct NetworkServiceConfig {
+    pub local_keys: identity::Keypair,
+    pub senders: Senders,
+    pub receivers: Receivers,
+    pub peerlist: PeerList,
+}
+
+#[derive(Clone)]
+pub struct Senders {
+    pub events: EventSender,
+    pub internal_events: InternalEventSender,
+    pub internal_commands: CommandSender,
+}
+
+pub struct Receivers {
+    pub commands: CommandReceiver,
+    pub internal_events: InternalEventReceiver,
+}
+
+type Shutdown = oneshot::Receiver<()>;
+
+pub mod integrated {
+    use super::*;
+
+    use bee_runtime::{node::Node, worker::Worker};
+
+    use async_trait::async_trait;
+
+    use std::{any::TypeId, convert::Infallible};
+
+    /// A node worker, that deals with processing user commands, and publishing events.
+    /// NOTE: This type is only exported to be used as a worker dependency.
+    #[derive(Default)]
+    pub struct NetworkService {}
+
+    #[async_trait]
+    impl<N: Node> Worker<N> for NetworkService {
+        type Config = NetworkServiceConfig;
+        type Error = Infallible;
+
+        fn dependencies() -> &'static [TypeId] {
+            vec![].leak()
```

Comment:
This array is empty so you can just use `&[]` instead of `vec![].leak()`. Pulling ZSTs and `'static` references to ZSTs out of thin air is permitted!

Reply:
done